src/index.ts

```diff
@@ -90,51 +103,42 @@ export default class ElectronGoogleOAuth2 extends EventEmitter {
    * @returns {Promise<string>}
    */
   openAuthWindowAndGetAuthorizationCode(urlParam: string) {
-    return new Promise<string>((resolve, reject) => {
-      const win = new BW({
-        useContentSize: true,
-        fullscreen: false,
-      });
+    return this.openAuthPageAndGetAuthorizationCode(urlParam);
+  }
 
-      win.loadURL(urlParam);
+  async openAuthPageAndGetAuthorizationCode(urlParam: string) {
+    if (this.server) {
+      // if a server is already running, we close it so that we free the port
+      // and restart the process
+      await this.server.close();
+      this.server = null;
+    }
+    this.server = new LoopbackRedirectServer({
+      port: LOOPBACK_INTERFACE_REDIRECTION_PORT,
+      callbackPath: '/callback',
+      successRedirectURL: this.successRedirectURL,
+    });
 
-      win.on('closed', () => {
-        reject(new UserClosedWindowError());
-      });
+    shell.openExternal(urlParam);
 
-      function closeWin() {
-        win.removeAllListeners('closed');
-        setImmediate( () => {
-          win.close();
-        });
-      }
-
-      win.webContents.on('did-navigate', (_event, newUrl) => {
-        const parsed = url.parse(newUrl, true);
-        if (parsed.query.error) {
-          reject(new Error(parsed.query.error_description as string));
-          closeWin();
-        } else if (parsed.query.code) {
-          resolve(parsed.query.code as string);
-          closeWin();
-        }
-      });
+    const reachedCallbackURL = await this.server.waitForRedirection();
 
-      win.on('page-title-updated', () => {
-        setImmediate(() => {
-          const title = win.getTitle();
-          if (title.startsWith('Denied')) {
-            reject(new Error(title.split(/[ =]/)[2]));
-            closeWin();
-          } else if (title.startsWith('Success')) {
-            resolve(title.split(/[ =]/)[2]);
-            closeWin();
-          }
-        });
-      });
-    });
-  }
+    // waitForRedirection will close the server
+    this.server = null;
 
+    const parsed = url.parse(reachedCallbackURL, true);
+    if (parsed.query.error) {
+      throw new Error(parsed.query.error_description as string);
+    } else if (!parsed.query.code) {
+      throw new Error('Unknown' as string);
+    }
+
+    // refocus on the window
+    BW.getAllWindows().filter(w => w.isVisible()).forEach(w => w.show());
```

Comment:
Should we put this logic behind an option, like `focusAfterSuccess: boolean = true`?

Reply:
Or trigger a custom event that we can listen on

Reply:
ok for the `focusAfterSuccess` option